### DISCUSSION TITLE
Open a tab instead of a window when kitty is running

### DISCUSSION THREAD

**Discussion by ZeroDeltaAlpha** [kovidgoyal/kitty#4929] [Q&A] [answered]
H Ali,

I've been trying to achieve the following behavior in kitty. In terminator you can pass the `--new-tab` argument which does the following actions
if terminator is already running it opens a new tab in the running terminal else open a new terminal 

The reasoning behind this behavior is that i bind `Alt+Return` to open terminals in i3 and would like a new tab if kitty is already running, is there anyway to achieve this behavior ?

I have tried making a script to detect if kitty has a listening socket in the environment (KITTY_LISTEN_ON) but running this from i3 using exec seems to never pick up the environment variable, my guess that it is unset from the exec call.

Many Thanks

**Comment by page-down** [comment, marked as answer]
> ... but running this from i3 ... my guess that it is unset ...

This environment variable is never set, because the window manager is not started by kitty.

Since you want to control kitty from an external process to open new tabs, you need to use remote control.
You should specify the socket path via configuration file or parameter when starting kitty, then check if it exists and send the remote control command.

https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.listen_on
https://sw.kovidgoyal.net/kitty/remote-control/#cmdoption-kitty-to

**Reply by ZeroDeltaAlpha** [reply]
Hi @page-down,

i have something like this currently:

```bash
#!/bin/sh
if [[ $KITTY_LISTEN_ON ]]
then
 echo "Kitty is running"
 kitty @ --to ${KITTY_LISTEN_ON} launch --type=tab  --keep-focus fish
else
 echo "Starting new kitty"
 kitty
fi

```

and I call that from i3 but it always results in a new terminal, anything i am missing here ?

Thanks in advance

**Reply by page-down** [reply]
> if [[ $KITTY_LISTEN_ON ]] ...
> ... anything i am missing here ?

See my previous reply.

The socket path should be explicitly specified.

https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-listen-on

```shell
SOCK="$XDG_RUNTIME_DIR/mykitty.sock"
[ -e "$SOCK" ] && kitty @ --to "unix:$SOCK" launch --type=tab || kitty -o allow_remote_control=yes --listen-on="unix:$SOCK"
```

**Reply by ZeroDeltaAlpha** [reply]
@page-down that code works perfectly for what i need many thanks for solving that for me 🥇
